src/servers/grpc_server.cc

```diff
@@ -727,21 +734,24 @@ InferAllocatorPayload(
   // invoked.
   for (const auto& io : request_header.output()) {
     if (io.has_shared_memory()) {
-      TRTSERVER_SharedMemoryBlock* smb = nullptr;
-      RETURN_IF_ERR(smb_manager->Get(&smb, io.shared_memory().name()));
-
       if (alloc_payload->shm_map_ == nullptr) {
```

Comment:
Move this as late as possible, otherwise the RETURN_IF_ERR below will cause a leak.

Reply:
Agreed